pandas/tests/indexes/common.py

```diff
@@ -937,3 +937,14 @@ def test_empty(self):
         index = self.create_index()
         assert not index.empty
         assert index[:0].empty
+
+    @pytest.mark.parametrize('how', ['outer', 'inner', 'left', 'right'])
+    def test_join_self(self, how):
+        index = self.create_index()
+        joined = index.join(index, how=how)
+
+        if index is index.unique():
```

Comment:
Ah, that I can reproduce. It is a bit strange this only happens for TimedeltaIndex (might be indication of a bug)

You can also test that `index` and ``joined`` are equal instead of identical.